hibernate-spatial/src/main/java/org/hibernate/spatial/SpatialDialect.java

```diff
@@ -82,6 +82,24 @@
 	 */
 	public String getIsEmptySQL(String columnName, boolean isEmpty);
 
+	/**
+	 * Returns the SQL fragment when parsing an <code>GeometryTypeExpression</code>.
+	 *
+	 * @param columnName the geometry column
+	 *
+	 * @return The SQL fragment for the geometrytype function
+	 */
+	public String getGeometryTypeSQL(String columnName);
+
+	/**
+	 * Returns the geometry type name specific to the actual dialect.
+	 *
+	 * @param geometryType the geometry type
+	 * 
+	 * @return the geometry type name
+	 */
+	public String getGeometryTypeValue(GeometryType geometryType);
```

Comment:
I find this method name rather confusing. I think getNameFor(GeometryType) would be clearer.